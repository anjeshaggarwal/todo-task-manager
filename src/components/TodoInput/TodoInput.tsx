import React from 'react'

interface IProps {
    onAdd: (text: string) => void
}

const TodoInput = React.memo((props: IProps) => {
    console.log('render input')
    const inputRef = React.useRef<HTMLInputElement>(null)
    const onAddHandler = (event: any) => {
        event.preventDefault();
        if (inputRef && inputRef.current) {
            const inputString = inputRef.current.value
            props.onAdd(inputString)
            inputRef.current.value = "";
            inputRef.current.focus();
        }
    }

    return <form>
        <input data-testid="input-text" ref={inputRef} type="text" placeholder={"Enter your Task"} />
        &nbsp;&nbsp;
        <input data-testid="input-submit" type="submit" onClick={onAddHandler} value="Add" />
    </form>
})

export default TodoInput