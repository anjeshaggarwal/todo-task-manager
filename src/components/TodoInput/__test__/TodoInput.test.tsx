import ReactDOM from 'react-dom'
import TodoInput from './../TodoInput'
import { render, cleanup, fireEvent } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TodoInput onAdd={jest.fn} />, div)
})

it('renders correctly', () => {
    const { getByTestId, getByPlaceholderText } = render(<TodoInput onAdd={jest.fn} />)
    expect(getByTestId('input-text')).toBeTruthy()
    expect(getByTestId('input-submit')).toBeTruthy()
    expect(getByPlaceholderText('Enter your Task')).toBeTruthy()
})

it('match snapshot', () => {
    const element = renderer.create(<TodoInput onAdd={jest.fn} />).toJSON()
    expect(element).toMatchSnapshot()
})

it('function call on click', ()=>{
    const addFunction = jest.fn()
    const {getByTestId} = render(<TodoInput onAdd={addFunction}/>)
    fireEvent.click(getByTestId('input-submit'))
    expect(addFunction).toHaveBeenCalled()
})