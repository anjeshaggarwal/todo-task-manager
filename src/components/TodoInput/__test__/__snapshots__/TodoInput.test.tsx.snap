// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<form>
  <input
    data-testid="input-text"
    placeholder="Enter your Task"
    type="text"
  />
    
  <input
    data-testid="input-submit"
    onClick={[Function]}
    type="submit"
    value="Add"
  />
</form>
`;
